### PR TITLE
Bump dnanexus app 1.0.0-beta.7

### PR DESCRIPTION
**Pull request by pditommaso**
This PR addresses two critical issues in the current version 
- Fixes the issue with the resume mechanism, see #24
- Fixes the issue with Nextflow log file not being uploaded on pipeline failure
- Fixes the issue with the Nextflow task retry, which is now fully supported 
- Fixes an issue with pipeline hanging when trying to access task empty error file

### CODE REVIEW

File: app/src/dxapp.sh [hunk 1]
```diff
@@ -92,11 +92,13 @@ main() {
...
   if [ -f .command.log ]; then
-    dx upload .command.log --path "${cmd_log_file}" --brief --no-progress || true
+    dx upload .command.log --path "${cmd_log_file}" --brief --wait --no-progress || true
   else
     >&2 echo "Missing Nextflow .command.log file"
   fi
-  dx-jobutil-add-output exit_code "$ret" --class=int
+  # mark the job as successful in any case, real task
+  # error code is managed by nextflow via .exitcode file
+  dx-jobutil-add-output exit_code "0" --class=int
```
Comment (commandlinegirl) [inline app/src/dxapp.sh:101]: Would it be possible to add a mention in the documentation (app Readme) how it works (that subjobs will always succeed).u!

**Review by commandlinegirl — COMMENTED** [review]
Please create an unpublished app so we can have a look at it before publishing. Thank you!

File: app/src/dxapp.sh (hunk 1 above)
Comment (pditommaso) [reply app/src/dxapp.sh:101]: This is an internal implementation detail. The job will create a flag file to mark the success or failed completion. Nextflow uses this file to detect the job status and decide the action to be taken. Adding this in the README will just confuse the user.

**Comment by pditommaso** [comment]
Ok, i've added the changelog and uploaded the app 

```
INFO:dxpy:Created app app-G6P049j0Y8QJ199Z70P39yYY
INFO:dxpy:Uploaded app nextflow-dx/1.0.0-beta.7 (app-G6P049j0Y8QJ199Z70P39yYY) successfully
INFO:dxpy:You can publish this app with: dx publish nextflow-dx/1.0.0-beta.7
{"id": "app-G6P049j0Y8QJ199Z70P39yYY"}
```

**Review by commandlinegirl — APPROVED** [review]
LGTM, thank you!

**Comment by commandlinegirl** [comment]
Oh, a minor thing I've noticed is that the app meta data lacks citations. Please consider adding it in the next version. Thanks.

**Comment by commandlinegirl** [comment]
> Ok, i've added the changelog and uploaded the app
> 
> ```
> INFO:dxpy:Created app app-G6P049j0Y8QJ199Z70P39yYY
> INFO:dxpy:Uploaded app nextflow-dx/1.0.0-beta.7 (app-G6P049j0Y8QJ199Z70P39yYY) successfully
> INFO:dxpy:You can publish this app with: dx publish nextflow-dx/1.0.0-beta.7
> {"id": "app-G6P049j0Y8QJ199Z70P39yYY"}
> ```

Thank you. Please publish the app when you get a chance.

**Comment by pditommaso** [comment]
> Oh, a minor thing I've noticed is that the app meta data lacks citations. 

Do you mean this? 
http://github.com/seqeralabs/xpack-dnanexus/blob/ff643adf9c5837adf29c863ddb9b8f2f07401ea9/app/dxapp.json#L9-L9

**Comment by pditommaso** [comment]
Ok, thanks

**Pull request merged**